File: src/leetcode/python/jewels_and_stones.py
import unittest


def numJewelsInStones(J, S):
    """
    :type J: str
    :type S: str
    :rtype: int
    """
    unique_j = set(J)
    count = 0
    for s in S:
        if s in unique_j:
            count += 1
    return count


class Test(unittest.TestCase):

    def testNumJewelsInStones(self):
        self.assertEquals(numJewelsInStones("aA", "aAAbbbb"), 3)
        self.assertEquals(numJewelsInStones("z", "ZZ"), 0)



if __name__ == "__main__":
    unittest.main()

File: src/leetcode/ruby/jewels_and_stones.rb
require "Set"

# def num_jewels_in_stones(j, s)
#   unique_j = j.split.uniq.join
#   return s.count(unique_j)
# end


# def num_jewels_in_stones(j, s)
#   s.chars.count { |c| j.include?(c) }
# end

def num_jewels_in_stones(j, s)
  count = 0
  s.split("").each do |c|
    if j.include?(c)
      count += 1
    end
  end
  return count
end


RSpec.describe do
  describe "num_jewels_in_stones" do
    it "returns 3" do
      expect(num_jewels_in_stones("aA", "aAAbbbb")).to eq(3)
    end

    it "returns 0" do
      expect(num_jewels_in_stones("z", "ZZ")).to eq(0)
    end
  end
end
